demos/PDE_constrained_optimisation/PDE_constrained_field.py

```diff
@@ -117,6 +115,7 @@
 approximation = BoussinesqApproximation(Ra=1, kappa=2e-2)
 delta_t = 0.1
 energy_solver = EnergySolver(T, u, approximation, delta_t, ImplicitMidpoint)
+energy_solver.solver_parameters.pop('ksp_converged_reason')
```

Comment:
Replace with `del energy_solver.solver_parameters["ksp_converged_reason"]` if the value to be removed isn't needed for anything. Also, why does this need to be deleted? Is it a once off for this test or is it an attribute that shouldn't be created until later?

Reply:
Done.